moldesign/mathutils/align.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import print_function, absolute_import, division
+from future.builtins import *
+from future import standard_library
+standard_library.install_aliases()
+
+# Copyright 2017 Autodesk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+from math import sqrt
+from .. import units as u
+from ..utils import exports
+
+@exports
+def rmsd_align(src_points, dst_points):
+    """ Calculate the 4x4 transformation matrix that aligns a set of source points 
+        to a set of destination of points.
+
+        This function implements the method from Kabsch (Acta Cryst. (1978) A34, 827-828) that 
+        calculate the optimal rotation matrix that minimizes the root mean squared deviation between 
+        two paired sets of points.
+
+    Args:
+        src_points(numpy.ndarray): An Nx3 array of the 3D source points.
+        dst_points(numpy.ndarray): An Nx3 array of the 3D destination points.
+
+    Returns:
+        rmsd_error(Float): The root mean square deviation measuring the alignment error.
+        Xform (np.ndarray): The 4x4 array that transforms the source points to the destination points. 
+    """
+
+    # Calculate point centers. 
+    num_points = src_points.shape[0]
+    src_center = src_points.sum(axis=0) / num_points
+    dst_center = dst_points.sum(axis=0) / num_points
+
+    # Calculate correlation matrix. 
+    C = np.dot(np.transpose(dst_points[:]-dst_center), src_points[:]-src_center)
```

Comment:
1. Would prefer to name this something like`correlation_mat` just to improve readability.
1. The variable names should be lowercase - it makes life easier for various introspection tools